src/dataExplorer/components/ResultsPane.tsx

```diff
@@ -142,29 +138,43 @@ const ResultsPane: FC = () => {
     disabledTitleText = 'Select a measurement before running script'
   }
 
-  const download = async () => {
-    event('CSV Download Initiated')
-    const promise = basic(
-      text,
-      {
-        vars: rangeToParam(range),
-      },
-      {
-        rawBlob: true,
-        language,
-        bucket: selection.bucket,
-      }
-    )
-    setCancelId(promise.id)
+  const downloadByServiceWorker = () => {
+    try {
+      event('runQuery', {context: 'query experience'})
```

Comment:
should we differentiate this event from a regular query execution? Technically it's running a query, but it's running a query for a CSV download. I think we can either do something like

```ts
event('runQuery.downloadCSV', {context: 'query experience'})
```
or something like
```ts
event('runQuery', {context: 'query experience - download CSV'})
```